app/controllers/aspects_controller.rb

```diff
@@ -44,14 +44,23 @@ def new
   end
 
   def destroy
-    @aspect = current_user.aspects.where(:id => params[:id]).first
+    @aspect = current_user.aspects.where(id: params[:id]).first
 
     begin
-      @aspect.destroy
-      flash[:notice] = I18n.t 'aspects.destroy.success', :name => @aspect.name
+      if current_user.auto_follow_back && @aspect.id == current_user.auto_follow_back_aspect.id
+        @aspect.destroy
+        current_user.auto_follow_back = false
+        current_user.auto_follow_back_aspect = nil
+        current_user.save
```

Comment:
We can combine these into one statement: `current_user.update(auto_follow_back: false, auto_follow_back_aspect: nil)`